src/Cli/func/Helpers/PythonHelpers.cs

```diff
@@ -391,37 +442,96 @@ public static async Task<Stream> ZipToSquashfsStream(Stream stream)
             }
         }
 
-        private static async Task<bool> ArePackagesInSync(string requirementsTxt, string pythonPackages)
+        private static async Task<bool> ArePackagesInSync(string functionAppRoot, string pythonPackages, PythonDependencyManager dependencyManager)
         {
-            var md5File = Path.Combine(pythonPackages, $"{Constants.RequirementsTxt}.md5");
+            var md5FileName = GetDependencyMd5FileName(dependencyManager);
+            var md5File = Path.Combine(pythonPackages, md5FileName);
             if (!FileSystemHelpers.FileExists(md5File))
             {
                 return false;
             }
 
             var packagesMd5 = await FileSystemHelpers.ReadAllTextFromFileAsync(md5File);
-            var requirementsTxtMd5 = SecurityHelpers.CalculateMd5(requirementsTxt);
+            var currentMd5 = CalculateDependencyChecksum(functionAppRoot, dependencyManager);
+
+            return packagesMd5 == currentMd5;
+        }
+
+        private static string GetDependencyFileName(PythonDependencyManager dependencyManager)
+        {
+            return dependencyManager switch
+            {
+                PythonDependencyManager.Pip => Constants.RequirementsTxt,
+                PythonDependencyManager.Poetry => Constants.PyProjectToml,
+                PythonDependencyManager.Uv => $"{Constants.PyProjectToml} + {Constants.UvLock}",
```

Comment:
Does it make sense for the file name to be two file names in one string? `"pyproject.toml uv.lock"` - if we're seperating this later, doesn't it make more sense to use a delimeter like `,` = `"pyproject.toml,uv.lock"`